### ISSUE TITLE
Too many open files

### ISSUE THREAD

**Issue by schmijos** [bbqsrc/cucumber-rust#43]
I'm trying to run the default setup from the README with the following test set: https://pragprog.com/titles/jbtracer/source_code

On my Macbook (beta-x86_64-apple-darwin, rustc 1.41.0-beta.2 (965ce16f7 2020-01-14)) with **cucumber_rust 0.6.4** this fails with the following message at scenario number 125:

```
thread 'main' panicked at 'Failed to capture stderr: Os { code: 24, kind: Other, message: "Too many open files" }', src/libcore/result.rs:1188:5
stack backtrace:
   0: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
   1: core::fmt::write
   2: std::io::Write::write_fmt
   3: std::panicking::default_hook::{{closure}}
   4: std::panicking::default_hook
   5: std::panicking::rust_panic_with_hook
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::result::unwrap_failed
   9: core::result::Result<T,E>::expect
  10: cucumber_rust::panic_trap::PanicTrap<T>::run_quietly
  11: cucumber_rust::panic_trap::PanicTrap<T>::run
  12: cucumber_rust::Steps<W>::run_scenario
  13: cucumber_rust::Steps<W>::run_scenarios
  14: cucumber_rust::Steps<W>::run
  15: cucumber_rust::CucumberBuilder<W,O>::run
  16: cucumber_rust::CucumberBuilder<W,O>::command_line
  17: cucumber::main
  18: std::rt::lang_start::{{closure}}
  19: std::panicking::try::do_call
  20: __rust_maybe_catch_panic
  21: std::rt::lang_start_internal
  22: std::rt::lang_start
  23: main
```

Is this test set too big for Cucumber Rust?

**Comment by bbqsrc** [comment]
Do you have a github repository that I can use to reproduce this?

**Comment by schmijos** [comment]
Here's the example: https://github.com/schmijos/cucumber-rust-reprod
It fails when running `cargo test`

**Comment by bbqsrc** [comment]
Thanks! I'll try to find time over the weekend.

**Comment by bbqsrc** [comment]
@schmijos run `cargo test -- --nocapture` and the bug doesn't manifest.

The issue is in the crate `shh`. It wraps stdout and stderr to capture messages from panics. It currently isn't closing the duplicated fds for stdout and stderr.

I've fixed it in a fork (https://github.com/bbqsrc/shh) but when testing against my develop branch with async support I found that it breaks the progress bar!

Hopefully I'll come up with a solution that doesn't require `-- --nocapture` next week.
